lib/index.js

```diff
@@ -51,18 +51,22 @@ function renderSongs(data) {
        <p>Artist: ${song.artist_name}</p>
        <p>Genre: ${veriGenre}</p>
        <p>Rating: ${song.track_rating}</p>
-       <button type="button" name="button">
-        <i class="far fa-star"></i>
-       </button>
    `;
+   const favHTML = `
+    <i class="far fa-star" id= "fav${index}"></i>`;
 
    const songDiv = document.createElement('div')
    songDiv.setAttribute('class', 'song');
    songDiv.setAttribute('id', `song-${index}`);
-   songDiv.addEventListener('click',function() {
-     addFavorite(index)});
    songDiv.innerHTML = songHTML;
    songsDiv.appendChild(songDiv);
+   const favBtn = document.createElement('button')
+   favBtn.setAttribute('class', 'favorite');
+   favBtn.setAttribute('id', `fav-${index}`);
+   favBtn.innerHTML = favHTML;
+   favBtn.addEventListener('click',function() {
+     addFavorite(index)});
+   songDiv.appendChild(favBtn)
   }
```

Comment:
You may be able to do this in one go by creating the HTML for it and setting it directly - rather than appending